### ISSUE TITLE
Feature: Option to insert styles above DOM elements vs head.

### ISSUE THREAD

**Issue by kmcaloon** [webpack-contrib/mini-css-extract-plugin#212]
Was wondering if there are plans or maybe already a way to implement this feature.  style-loader is configurable so that we can inject styles directly above their relevant components by setting the following option: `{ insertAt: top }`.  Unless I am missing something, one main advantage of such an approach seems to be a more optimal rendering on initial page load as DOM elements will only be blocked the loading of necessary styles vs those for everything on the page.  This is somewhat related to Pagespeed's recommendations for inlining small stylesheets throughout the body.  

Would this make sense?

**Comment by aleen42** [comment]
Is there any consideration?

**Comment by evilebottnawi** [comment]
How it is related to `mini-css-extract-plugin`, we just extract css to separate file and all, youself solve where include `link` tag

**Comment by aleen42** [comment]
It seems reasonable as I can resolve it by setting the position of entries.

**Comment by AlexDubok** [comment]
> How it is related to `mini-css-extract-plugin`, we just extract css to separate file and all, youself solve where include `link` tag

Yeah, but somehow this plugin has the default behavior of appending extracted files to the head :)
https://github.com/webpack-contrib/mini-css-extract-plugin/blob/master/src/index.js#L367

**Comment by Birowsky** [comment]
Hold up, which plugin has the ordering responsibility?

**Comment by prateekbh** [comment]
Hi! This plugin still seems to insert the dynamically loaded chunks in head

https://github.com/webpack-contrib/mini-css-extract-plugin/blob/master/src/index.js#L379

Is there a way that these dynamically loaded styles can be put in some other place? e.g. end of body?
